syntax = "proto3";

message OrderMSG {
  int64 originId = 1;
  string symbol = 3;
  bool buySide = 2; 
  int64 quantitiy = 4;
  int64 price = 5;
}

message OrderFillMSG {
  int64 filled = 2;
}

message OrderResponseMSG {
  bool successful = 1;
  int64 amountFilled = 2;
  int64 price = 3;
}

message SignupMSG{
  string name = 1;
}

enum SignupResponseEnum {
  success = 0;
  tooLong = 1;
  taken = 2;
}

message SignupResponseMSG {
  SignupResponseEnum result = 1;
}